Modules/_sha3/sha3module.c

```diff
@@ -746,8 +735,26 @@ PyInit__sha3(void)
         goto error;
     }
 
-    return m;
+    return 0;
   error:
-    Py_DECREF(m);
-    return NULL;
+    return -1;
```

Comment:
IMO you can just remove the error label and replace "goto error" with "return -1".